gtwrap/matlab_wrapper.py

```diff
@@ -1518,7 +1519,11 @@ def generate_wrapper(self, namespace):
         ptr_ctor_frag = ''
 
         for cls in self.classes:
-            if cls.cpp_class().strip() in self.ignore_classes:
+            uninstantiated_name = "::".join(cls.namespaces()[1:]) + "::" + cls.name
+            self._debug("Cls: {} -> {}".format(cls.name, uninstantiated_name))
+
+            if uninstantiated_name in self.ignore_classes:
+                self._debug("Ignoring: {} -> {}".format(cls.name, uninstantiated_name))
```

Comment:
A little context about what's going on here? :slightly_smiling_face:

Reply:
The original way of ignoring classes will not work for templated classes, for example:
```
typedef gtsam::BearingRangeFactor<A,B,C,D> BearingRangeFactor2D;
```

I added an ignore for gtsam to unbreak the MATLAB wrapper compilation, but eventually we have to add this feature. This is caused by the template instantiator not replacing type arguments in the return types of the member functions.

Reply:
Note it's not an issue for pybind as pybind does not have return type in generated C++.

Reply:
Gotcha. This should technically be another PR but I guess it's fine.